### PR TITLE
Simplify the notification schema for the snooze work.

### PR DESCRIPTION
**Pull request by nickmcmahon01**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/cmd/api/service/NotificationService.kt [hunk 1]
```diff
@@ -19,70 +19,77 @@ import java.util.*
 class NotificationService(@Autowired val shiftNotificationRepository: ShiftNotificationRepository, @Autowired val shiftTaskNotificationRepository: ShiftTaskNotificationRepository, @Autowired val clock: Clock, @Autowired val authenticationFacade: AuthenticationFacade, @Value("\${notifications.to.defaultMonths}") var monthStep: Long) {
 
     @Transactional
-    fun getNotifications(unreadOnlyParam: Optional<Boolean>, fromParam: Optional<LocalDate>, toParam: Optional<LocalDate>): Collection<NotificationDto> {
+    fun getNotifications(unacknowledgedOnlyParam: Optional<Boolean>, fromParam: Optional<LocalDate>, toParam: Optional<LocalDate>): Collection<NotificationDto> {
```
Comment (jonhindle) [inline src/main/java/uk/gov/justice/digital/hmpps/cmd/api/service/NotificationService.kt:22]: Minor point is 'Param' in the argument name necessary.

Comment (jonhindle) [reply src/main/java/uk/gov/justice/digital/hmpps/cmd/api/service/NotificationService.kt:22]: ah, you use it again down there...

File: src/main/java/uk/gov/justice/digital/hmpps/cmd/api/dto/NotificationDto.kt [hunk 2]
```diff
@@ -17,8 +17,8 @@ data class NotificationDto @JsonCreator constructor(
         @JsonProperty("lastModified")
         val lastModified: LocalDateTime,
         @ApiModelProperty(required = true, value = "Whether the notification has been acknowledged", position = 3, example = "true")
-        @JsonProperty("read")
-        val read: Boolean
+        @JsonProperty("acknowledged")
+        val acknowledged: Boolean
```
Comment (astroash) [inline src/main/java/uk/gov/justice/digital/hmpps/cmd/api/dto/NotificationDto.kt:21]: Would `seen` be a simpler word here?

File: src/main/resources/db/migration/general/V4_0__modify_notification_add_snooze.sql [hunk 3]
```diff
@@ -0,0 +1,52 @@
+ALTER TABLE "SHIFT_NOTIFICATION"
```
Comment (astroash) [inline src/main/resources/db/migration/general/V4_0__modify_notification_add_snooze.sql:1]: I know we changed the casing on the table on last PR, but having table names and columns capitalised makes SQL harder to read as there's less visual queues between key words and named stuff.

Not saying this needs to be changed, just sharing thoughts

File: src/main/resources/db/migration/general/V4_0__modify_notification_add_snooze.sql [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    DROP COLUMN "READ";
+
+ALTER TABLE "SHIFT_NOTIFICATION"
+    DROP COLUMN "SENT_SMS";
+
+ALTER TABLE "SHIFT_NOTIFICATION"
+    DROP COLUMN "SENT_EMAIL";
+
+ALTER TABLE "SHIFT_NOTIFICATION"
+    DROP COLUMN "DATE_TIME";
```
Comment (astroash) [inline src/main/resources/db/migration/general/V4_0__modify_notification_add_snooze.sql:20]: Out of curiousity, what was this column for. Was it for when the notification was created?

File: src/main/java/uk/gov/justice/digital/hmpps/cmd/api/dto/NotificationDto.kt (hunk 2 above)
Comment (nickmcmahon01) [reply src/main/java/uk/gov/justice/digital/hmpps/cmd/api/dto/NotificationDto.kt:21]: its 'has been seen on the UI, emailed out, sms'd out or ignored' I'm up for a shorter word though

File: src/main/resources/db/migration/general/V4_0__modify_notification_add_snooze.sql (hunk 4 above)
Comment (nickmcmahon01) [reply src/main/resources/db/migration/general/V4_0__modify_notification_add_snooze.sql:20]: It's the day the Sql was run, so in the query its just SELECT SYS_DATE, etc
https://github.com/ministryofjustice/cmd-csr-procs/blob/master/Notification_Shift/GET_MODIFIED_SHIFTS.sql#L52
It'll be to make notifications unique in the database, but its not an audit of changes, its a scratch database so I don't care if the shift changes twice before we get a chance to send the notificaiton, we only want to send the latest

File: src/main/java/uk/gov/justice/digital/hmpps/cmd/api/dto/NotificationDto.kt (hunk 2 above)
Comment (jonhindle) [reply src/main/java/uk/gov/justice/digital/hmpps/cmd/api/dto/NotificationDto.kt]: `seen` is not much different to `read`

**Pull request merged**